sync/sync-api/src/main/java/com/duckduckgo/sync/api/DeviceSyncState.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.sync.api
+
+/** Public interface for Device Sync State */
+interface DeviceSyncState {
+    /**
+     * Checks if feature should be visible to users.
+     */
+    fun isFeatureEnabled(): Boolean
+
+    /**
+     * Checks if user is signed in on this device.
+     */
+    fun isDeviceSyncEnabled(): Boolean
```

Comment:
nit: naming. The description and the method name are not consistent. Either call it `isSignedIn` or change the description

Reply:
sure thing. renamed to `isUserSignedInOnDevice`.